lib/gateway/make-resolver.js

```diff
@@ -38,7 +38,32 @@ function getDirectiveSelection (node, directiveName) {
   return query.definitions[0].selectionSet.selections
 }
 
-function removeNonServiceTypeFields (selections, service, type, schema) {
+function getDirectiveRequiresSelection (selections, type) {
+  if (!type.extensionASTNodes ||
+    !type.extensionASTNodes[0].fields[0] ||
+    !type.extensionASTNodes[0].fields[0].directives[0]) {
+    return []
+  }
+
+  const requires = []
+  const selectedFields = selections.map(selection => selection.name.value)
+
+  for (const node of type.extensionASTNodes) {
+    for (const field of node.fields) {
+      if (!selectedFields.includes(field.name.value)) {
+        continue
+      }
+      const directive = field.directives.find(d => d.name.value === 'requires')
```

Comment:
can this be null for some reason?

Reply:
I'm writing a test case

Reply:
can't find a case. help?

Reply:
Looking at the logic, any non-external field containing a directive that isn't `requires` should get you the test case.

For example:

Schema:
```graphql
directive @custom on OBJECT | FIELD_DEFINITION

type Size @key(fields: "id") {
  id: Int!
  cpus: Int!
  memory: Int!
}

type Region @key(fields: "id") {
  id: Int!
  city: String!
}

type Metadata @key(fields: "id") {
  id: Int!
  description: String!
}

extend type Host @key(fields: "id") {
  id: Int! @external
  size: Int! @external
  region: Int! @external
  # Field defined here with directive that isn't `requires`
  metadata: Metadata! @custom
  sizeData: Size! @requires(fields: "size")
  regionData: Region! @requires(fields: "region")
}
```

Query:
```graphql
query {
  hosts {
    name
    sizeData {
      cpus
    }
    metadata {
      description
    }
  }
}
```

Reply:
many thanks @jonnydgreen !